esphome/components/online_image/__init__.py

```diff
@@ -0,0 +1,161 @@
+import logging
+
+from esphome import automation
+import esphome.codegen as cg
+from esphome.components.http_request import CONF_HTTP_REQUEST_ID, HttpRequestComponent
+from esphome.components.image import (
+    CONF_USE_TRANSPARENCY,
+    IMAGE_TYPE,
+    Image_,
+    validate_cross_dependencies,
+)
+import esphome.config_validation as cv
+from esphome.const import (
+    CONF_BUFFER_SIZE,
+    CONF_FORMAT,
+    CONF_ID,
+    CONF_ON_ERROR,
+    CONF_RESIZE,
+    CONF_TRIGGER_ID,
+    CONF_TYPE,
+    CONF_URL,
+)
+
+DEPENDENCIES = ["display"]
+AUTO_LOAD = ["http_request", "image"]
+CODEOWNERS = ["@guillempages"]
```

Comment:
`http_request` may need its own configuration so it should not be auto laoded.
```suggestion
AUTO_LOAD = ["image"]
DEPENDENCIES = ["display", "http_request"]
CODEOWNERS = ["@guillempages"]
```

Reply:
Well, the idea was that if the default configuration works, there is no need to explicitly list the dependency, and only if there is something extra to do (like disabling SSL on arduino) you then explicitly add the dependency. I documented it also this way in the docs PR.
But it's not a big deal either way, and it might be clearer (and most important: consistent with the rest of the code base) to make it an explicit dependency.